client/consensus/slots/src/lib.rs

```diff
@@ -242,6 +246,38 @@ pub trait SimpleSlotWorker<B: BlockT> {
 		Some(proposal)
 	}
 
+	/// Calls `create_inherent_data` and handles errors.
+	async fn create_inherent_data(
+		slot_info: &SlotInfo<B>,
+		logging_target: &str,
+	) -> Option<sp_inherents::InherentData> {
+		let inherent_data = match slot_info.create_inherent_data.create_inherent_data().await {
```

Comment:
dumb question: now that this method is async, why not do something like `create_inherent_data().timeout(timeout).await` instead of calculating the duration after the fact?